rust/rope/benches/diff.rs

```diff
@@ -19,95 +19,95 @@ extern crate xi_rope;
 
 use test::Bencher;
 use xi_rope::compare;
+use xi_rope::Rope;
 
 static EDITOR_STR: &str = include_str!("../../core-lib/src/editor.rs");
 static VIEW_STR: &str = include_str!("../../core-lib/src/view.rs");
 
 static INTERVAL_STR: &str = include_str!("../src/interval.rs");
 static BREAKS_STR: &str = include_str!("../src/breaks.rs");
 
+fn make_test_data() -> (Vec<u8>, Vec<u8>) {
+    let one = [EDITOR_STR, VIEW_STR, INTERVAL_STR, BREAKS_STR].concat().into_bytes();
+    let mut two = one.clone();
+    let idx = one.len() / 2;
+    two[idx] = 0x02;
+    (one, two)
+}
+
 #[bench]
 fn ne_idx_sw(b: &mut Bencher) {
-    let one: String = [EDITOR_STR, VIEW_STR, INTERVAL_STR, BREAKS_STR].concat();
-    let mut two = one.clone();
-    unsafe {
-        let b = two.as_bytes_mut();
-        let idx = b.len() - 200;
-        b[idx] = 0x02;
-    }
+    let (one, two) = make_test_data();
 
     b.iter(|| {
-        compare::ne_idx_fallback(one.as_bytes(), one.as_bytes());
-        compare::ne_idx_fallback(one.as_bytes(), two.as_bytes());
+        compare::ne_idx_fallback(&one, &one);
+        compare::ne_idx_fallback(&one, &two);
     })
 }
 
 #[bench]
-fn ne_idx_hw(b: &mut Bencher) {
-    let one: String = [EDITOR_STR, VIEW_STR, INTERVAL_STR, BREAKS_STR].concat();
-    let mut two = one.clone();
-    unsafe {
-        let b = two.as_bytes_mut();
-        let idx = b.len() - 200;
-        b[idx] = 0x02;
-    }
+fn ne_idx_sse(b: &mut Bencher) {
+    let (one, two) = make_test_data();
```

Comment:
Do we want to `if is_x86_feature_detected("sse4.2")` around this so that it won't crash when benchmarking on ancient systems?